src/ArrowFunctions.cpp

```diff
@@ -311,61 +312,44 @@ int64_t cpp_getListColumnSize(const char* filename, const char* colname, void* c
           while (int_reader->HasNext()) {
             int64_t value;
             (void)int_reader->ReadBatch(1, &definition_level, &rep_lvl, &value, &values_read);
-            if (values_read == 0){ // empty segment
-              if (!int_reader->HasNext()){ // last value 
-                offsets[i] = seg_size;
-              }
-              else {
-                i++;
-              }
-            } else {
-              // new segment starts. This has to be before increment to avoid false empty on first segment.
-              if (rep_lvl == 0 && seg_size >0) {
-                offsets[i] = seg_size;
-                i++;
-                seg_size = 0;
-              }
-              // increment counters
+            if (values_read == 0 || (!first && rep_lvl == 0)) {
+              offsets[i] = seg_size;
```

Comment:
should we rename `offsets` in this function since it's really the `seg_sizes`

Reply:
We could, but we do the same thing in the strings case. I think it makes more sense though, so I will update it.

Reply:
I don't think it really matters since we do the same thing in the Strings case. I can update it though so it is clearer.

Reply:
I updated the entire workflow instead of just this function.